contracts/stellar-interchain-token-service/src/token_metadata.rs

```diff
@@ -59,26 +59,29 @@ pub fn token_metadata(
         .try_decimals()
         .map_err(|_| ContractError::InvalidTokenAddress)?
         .map_err(|_| ContractError::TokenInvocationError)?;
-    let name = token
```

Comment:
the optimization isn't particularly important, since deployment of native token only occurs once. still, we can add it as long as it's done in a clean way below